packages/console/src/ds-components/KeyValueInputField/index.tsx

```diff
@@ -0,0 +1,121 @@
+import { useCallback } from 'react';
+import { type FieldError } from 'react-hook-form';
+
+import CirclePlus from '@/assets/icons/circle-plus.svg';
+import Minus from '@/assets/icons/minus.svg';
+import Button from '@/ds-components/Button';
+import IconButton from '@/ds-components/IconButton';
+import TextInput, { type Props as TextInputProps } from '@/ds-components/TextInput';
+
+import * as styles from './index.module.scss';
+
+type FieldType = {
+  key: string;
+  value: string;
+};
+
+type ErrorType = {
+  [K in keyof FieldType]?: FieldError | string | undefined;
+};
+
+// TextInput props getter
+type InputFieldPropsGetter = {
+  [K in keyof FieldType]: (index: number) => Omit<TextInputProps, 'ref'>;
+};
+
+type Props = {
+  className?: string;
+  fields: Array<FieldType & { id: string }>; // Id is required to uniquely identify each field
+  errors?: Array<ErrorType | undefined>;
+  getInputFieldProps: InputFieldPropsGetter;
+  onRemove: (index: number) => void;
+  onAppend: (field: FieldType) => void;
+};
+
+/**
+ * UI component for key-value input field.
+ *
+ * This component is used to add multiple key-value pairs.
+ * For most of the cases, it is designed to be used along with react-hook-form.
+ * All the input properties are registered with react-hook-form.
+ * @param {Props} props - The props for the component.
+ * @param {string} [props.className] - The class name for the container.
+ * @param {FieldType} props.fields - The array of key-value pairs. @see {@link https://react-hook-form.com/docs/usefieldarray}
+ * @param {ErrorType[]} [props.errors] - The array of errors for each field. Accepts both string and FieldError from RHF.
+ * @param {Function} props.onRemove - The function to remove a field. @see {@link https://react-hook-form.com/docs/usefieldarray}
+ * @param {Function} props.onAppend - The function to append a new field. @see {@link https://react-hook-form.com/docs/usefieldarray}
+ * @param {InputFieldPropsGetter} getInputFieldProps - The function bundle to get the input field props for each field. e.g. Use React Hook Form's register method to register the input field.
```

Comment:
Why do you use a getter function, instead of passing the field props as a param?

Reply:
Because the props value is dynamic, it is calculated based on the index input.